baseapp/saml/serviceprovider.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2019 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package saml
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/RobotsAndPencils/go-saml"
```

Comment:
jc: why this lib instead of https://github.com/crewjam/saml? go-saml looks a little unmaintained, and it looks like we don't use this lib anywhere internally, but we do use crewjam for a variety of other applications.